File: kobciye/lib/features/dashboards/student/student_dashboard.dart
import 'package:flutter/material.dart';
import '../../../core/constants/app_colors.dart';
import '../../../core/widgets/dashboard_shell.dart';
import '../../../core/widgets/stat_card.dart';
import '../../../models/app_user.dart';
import '../widgets/highlight_banner.dart';
import '../widgets/overview_page.dart';
import '../widgets/section_placeholder.dart';

/// Student-facing home for [AppRole.student].
///
/// Order must mirror `NavItems.forRole(AppRole.student)`:
/// Dashboard, Attendance, Results, Progress, Notices.
class StudentDashboard extends StatelessWidget {
  final AppUser user;

  const StudentDashboard({super.key, required this.user});

  @override
  Widget build(BuildContext context) {
    return DashboardShell(
      user: user,
      roleLabelKey: 'role_student',
      pages: [
        OverviewPage(
          titleKey: 'overview',
          stats: const [
            StatCard(label: 'My attendance', value: '97%', icon: Icons.fact_check_rounded, tint: AppColors.success, trend: 'this term', trendUp: true),
            StatCard(label: 'Latest result', value: 'A-', icon: Icons.grading_rounded, tint: AppColors.primaryLight, trend: 'Mathematics', trendUp: true),
            StatCard(label: 'Payment status', value: 'Paid', icon: Icons.payments_rounded, tint: AppColors.success, trend: 'up to date', trendUp: true),
            StatCard(label: 'Qur\'an progress', value: 'Juz\' 7', icon: Icons.menu_book_rounded, tint: AppColors.accent, trend: 'memorizing', trendUp: true),
          ],
          highlight: const HighlightBanner(
            title: 'Keep up the great work! 🌟',
            message: 'You\'ve had perfect attendance for 3 weeks in a row — your teacher left you an encouraging note.',
            icon: Icons.emoji_events_rounded,
            gradient: AppColors.goldGradient,
          ),
        ),
        const SectionPlaceholder(titleKey: 'attendance', icon: Icons.fact_check_rounded, gradient: AppColors.brandGradient),
        const SectionPlaceholder(titleKey: 'results', icon: Icons.grading_rounded, gradient: AppColors.growthGradient),
        const SectionPlaceholder(titleKey: 'progress', icon: Icons.trending_up_rounded, gradient: AppColors.goldGradient),
        const SectionPlaceholder(titleKey: 'notices', icon: Icons.campaign_rounded, gradient: AppColors.riskMediumGradient),
      ],
    );
  }
}

File: kobciye-rn/src/screens/dashboards/StudentDashboard.js
import React from 'react';
import { Colors, Gradients } from '../../constants/colors';
import DashboardShell from '../../widgets/DashboardShell';
import OverviewPage from '../../widgets/OverviewPage';
import StatCard from '../../widgets/StatCard';
import HighlightBanner from '../../widgets/HighlightBanner';
import SectionPlaceholder from '../../widgets/SectionPlaceholder';
import AttendanceSection from '../../widgets/AttendanceSection';
import ExamsSection from '../../widgets/ExamsSection';
import MessagingSection from '../../widgets/MessagingSection';
import TeacherProfileCard from '../../widgets/TeacherProfileCard';
import { View, Text, ScrollView, StyleSheet } from 'react-native';
import { Text as TextStyles, Spacing } from '../../constants/text';

function MyTeacherPage() {
  return (
    <ScrollView contentContainerStyle={styles.wrap} showsVerticalScrollIndicator={false}>
      <View style={{ gap: 4 }}>
        <Text style={TextStyles.h1}>My teacher</Text>
        <Text style={TextStyles.bodyMuted}>Your class teacher and subject teachers — message them anytime through Kobciye.</Text>
      </View>
      <TeacherProfileCard
        name="Teacher Amina Yusuf"
        subject="Mathematics"
        responsibility="Class teacher"
        className="Grade 5 - A"
      />
      <TeacherProfileCard
        name="Teacher Sahra Maxamed"
        subject="Science"
        responsibility="Subject teacher"
        className="Grade 5 - A"
      />
      <TeacherProfileCard
        name="Teacher Cabdiraxman Cali"
        subject="English"
        responsibility="Subject teacher"
        className="Grade 5 - A"
      />
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  wrap: { gap: Spacing.md, paddingBottom: Spacing.xl },
});

// Order mirrors navItemsForRole(student): Dashboard, Attendance, Exams, Results, My Teacher, Messages, Notices.
export default function StudentDashboard({ route }) {
  const user = route.params.user;

  return (
    <DashboardShell
      user={user}
      roleLabelKey="role_student"
      pages={[
        <OverviewPage
          titleKey="overview"
          stats={[
            <StatCard label="My attendance" value="97%" icon="checkbox" tint={Colors.success} trend="this term" trendUp />,
            <StatCard label="Latest result" value="A-" icon="ribbon" tint={Colors.primaryLight} trend="Mathematics" trendUp />,
            <StatCard label="Payment status" value="Paid" icon="card" tint={Colors.success} trend="up to date" trendUp />,
          ]}
          highlight={
            <HighlightBanner
              title="Keep up the great work! 🌟"
              message="You've had perfect attendance for 3 weeks in a row — your teacher left you an encouraging note."
              icon="trophy"
              gradient={Gradients.gold}
            />
          }
        />,
        <AttendanceSection title="My attendance" />,
        <ExamsSection />,
        <SectionPlaceholder titleKey="results" icon="ribbon" gradient={Gradients.growth} />,
        <MyTeacherPage />,
        <MessagingSection />,
        <SectionPlaceholder titleKey="notices" icon="megaphone" gradient={Gradients.riskMedium} />,
      ]}
    />
  );
}
